pipeline/compute_metrics.py

```diff
@@ -138,11 +139,39 @@ def fill_up_missing(self):
     def is_incomplete(self, elem):
         incomplete = 'loose_eval' not in elem or 'strict_eval' not in elem
         if incomplete:
-            return False
+            return True
         return (
-            'follow_all_instructions' not in elem['loose_eval'] or 'follow_all_instructions' not in elem['strict_eval']
+            'follow_instruction_list' not in elem['loose_eval']
+            or 'follow_instruction_list' not in elem['strict_eval']
+            and 'instruction_id_list' not in elem['loose_eval']
+            or 'instruction_id_list' not in elem['strict_eval']
```

Comment:
This is a complex clause imo. There is "or" and "and" mixed in without parenthesis which can make it hard to parse the true condition. I think you're trying to cover for loose and strict eval conditions.  

This might be a cleaner way to get to it:

if 'follow_instruction_list' in elem['loose_eval'] and 'instruction_id_list' in elem['loose_eval']:
     return False
if 'follow_instruction_list' in elem['strict_eval'] and 'instruction_id_list' in elem['strict_eval']:
     return False

return True

Reply:
thanks, in fact there were even more mistakes related to that logic, hopefully it's correct now